libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -156,55 +156,53 @@ void ContractLevelChecker::findDuplicateDefinitions(map<string, vector<T>> const
 	}
 }
 
-void ContractLevelChecker::checkAbstractFunctions(ContractDefinition const& _contract)
+void ContractLevelChecker::checkAbstractDefinitions(ContractDefinition const& _contract)
 {
-	// Mapping from name to function definition (exactly one per argument type equality class) and
-	// flag to indicate whether it is fully implemented.
-	using FunTypeAndFlag = std::pair<FunctionTypePointer, bool>;
-	map<string, vector<FunTypeAndFlag>> functions;
+	// Collects functions, static variable getters and modifiers. If they
+	// override (unimplemented) base class ones, they are replaced.
+	set<OverrideProxy, OverrideProxy::CompareBySignature> proxies;
 
-	auto registerFunction = [&](Declaration const& _declaration, FunctionTypePointer const& _type, bool _implemented)
+	auto registerProxy = [&proxies](OverrideProxy const& _overrideProxy)
 	{
-		auto& overloads = functions[_declaration.name()];
-		auto it = find_if(overloads.begin(), overloads.end(), [&](FunTypeAndFlag const& _funAndFlag)
+		auto it = proxies.find(_overrideProxy);
```

Comment:
While iterators might be faster, code that uses them is usually hard to read and easy to get wrong.
The following code has twice the complexity but is hopefully easier to understand:
```
// Remove any "equal" proxies that might be unimplemented.
if (!_overrideProxy.unimplemented())
    proxies.erase(_overrideProxy);
proxies.insert(_overrideProxy);
```

Reply:
I believe the iterator code is O(1), the position for insertion is exactly given, and a normal insert is log(size). I agree your code is more readable. I'll rewrite.